hbase-client/src/main/java/org/apache/hadoop/hbase/client/RawAsyncTableImpl.java

```diff
@@ -944,4 +1001,59 @@ public <S, R> CoprocessorServiceBuilder<S, R> coprocessorService(
     PartialResultCoprocessorCallback<S, R> callback) {
     return new CoprocessorServiceBuilderImpl<>(stubMaker, callable, callback);
   }
+
+  private static class MultiRegionCoprocessorServiceProgress<R>{
```

Comment:
The state needed to manage the all the individual RPCs to regions was being passed around as parameters, this moves them into this helper class along with some additional state